common/viewer.cpp

```diff
@@ -83,7 +101,203 @@ namespace rs2
+    const rs2_option save_to_ply::OPTION_PLY_BINARY;
+    const rs2_option save_to_ply::OPTION_PLY_NORMALS;
+
+    void viewer_model::set_export_popup(ImFont* large_font, ImFont* font, rect stream_rect, std::string& error_message, config_file& temp_cfg)
+    {
+        float w = 520; // hardcoded size to keep popup layout
+        float h = 325;
+        float x0 = stream_rect.x + stream_rect.w / 3;
+        float y0 = stream_rect.y + stream_rect.h / 3;
+        ImGui::SetNextWindowPos({ x0, y0 });
+        ImGui::SetNextWindowSize({ w, h });
+
+        auto flags = ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove |
+            ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoSavedSettings;
+
+        ImGui_ScopePushFont(font);
+        ImGui::PushStyleColor(ImGuiCol_PopupBg, sensor_bg);
+        ImGui::PushStyleColor(ImGuiCol_TextSelectedBg, white);
+        ImGui::PushStyleColor(ImGuiCol_Text, light_grey);
+        ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(15, 15));
+        ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 1);
+
+        static export_type tab = export_type::ply;
+        if (ImGui::BeginPopupModal("Export", nullptr, flags))
+        {
+            ImGui::SetCursorScreenPos({ (float)(x0), (float)(y0 + 30) });
+            ImGui::PushStyleColor(ImGuiCol_Button, sensor_bg);
+            ImGui::PushStyleColor(ImGuiCol_ButtonHovered, sensor_bg);
+            ImGui::PushFont(large_font);
+            for (auto& exporter : exporters)
+            {
+                ImGui::PushStyleColor(ImGuiCol_Text, tab != exporter.first ? light_grey : light_blue);
+                ImGui::PushStyleColor(ImGuiCol_TextSelectedBg, tab != exporter.first ? light_grey : light_blue);
+                ImGui::SameLine();
+                if (ImGui::Button(exporter.second.name.c_str(), { w / exporters.size() - 50, 30 }))
+                {
+                    config_file::instance().set(configurations::viewer::settings_tab, tab);
+                    temp_cfg.set(configurations::viewer::settings_tab, tab);
+                    tab = exporter.first;
+                }
+                ImGui::PopStyleColor(2);
+            }
+
+            ImGui::PopFont();
+            if (tab == export_type::ply)
+            {
+                bool mesh = temp_cfg.get(configurations::ply::mesh);
+                bool use_normals = temp_cfg.get(configurations::ply::use_normals);
+                if (!mesh) use_normals = false;
+                int encoding = temp_cfg.get(configurations::ply::encoding);
+
+                ImGui::PushStyleColor(ImGuiCol_Text, grey);
+                ImGui::Text("Polygon File Format defines a flexible systematic scheme for storing 3D data");
+                ImGui::PopStyleColor();
+                ImGui::NewLine();
+                ImGui::SetCursorScreenPos({ (float)(x0 + 15), (float)(y0 + 90) });
+                ImGui::Separator();
+                if (ImGui::Checkbox("Meshing", &mesh))
```

Comment:
The checkbox explanation and caption are not complete, especially what will be the output if button left unchecked. The button would be used as an addon "Generate Mesh" to generate Polygons in addition to Points (that always present)